src/components/profiles-modals/RenameProfileModal.vue

```diff
@@ -1,75 +1,55 @@
-<script lang="ts">
-import { Component, Ref, Vue, Watch } from 'vue-property-decorator';
+<script lang="ts" setup>
 import { ModalCard } from "../all";
 import R2Error from "../../model/errors/R2Error";
-import Profile from "../../model/Profile";
 import { useProfilesComposable } from '../composables/ProfilesComposable';
+import { computed, nextTick, ref, watchEffect } from 'vue';
+import { getStore } from '../../providers/generic/store/StoreProvider';
+import { State } from '../../store';
 
-@Component({
-    components: {ModalCard}
-})
-export default class RenameProfileModal extends Vue {
-    @Ref() readonly nameInput: HTMLInputElement | undefined;
-    private newProfileName: string = '';
-    private renamingInProgress: boolean = false;
-
-    get doesProfileExist() {
-        const { doesProfileExist } = useProfilesComposable();
-        return doesProfileExist;
-    }
+const store = getStore<State>();
 
-    get makeProfileNameSafe() {
-        const { makeProfileNameSafe } = useProfilesComposable();
-        return makeProfileNameSafe;
-    }
+const {
+    doesProfileExist,
+    makeProfileNameSafe,
+} = useProfilesComposable();
 
-    @Watch('$store.state.profile.activeProfile')
-    activeProfileChanged(newProfile: Profile, oldProfile: Profile|null) {
```

Comment:
This doesn't actually seem to affect anything. Removing has had no apparent behaviour changes.

Reply:
It's probably a leftover from when all the profile modals were in one component and some of the UI elements were shared (e.g. between creating and renaming a profile).